packages/gatsby/src/state-machines/query-running/index.ts

```diff
@@ -0,0 +1,78 @@
+import { MachineConfig, Machine } from "xstate"
+import { IQueryRunningContext } from "./types"
+import { queryRunningServices } from "./services"
+import { queryActions } from "./actions"
+
+export const queryStates: MachineConfig<IQueryRunningContext, any, any> = {
+  initial: `extractingQueries`,
+  states: {
+    extractingQueries: {
+      id: `extracting-queries`,
+      invoke: {
+        id: `extracting-queries`,
+        src: `extractQueries`,
+        onDone: [
+          {
+            actions: `resetGraphQLRunner`,
```

Comment:
This isn't correct behavior. Extracting queries shouldn't result in reset GraphQL Runner, graphql runner should be reset if data or schema changes.

Reply:
OK, I'll move it into the data layer so it runs after schema inference. This shouldn't be breaking though, right? It just means that it will be addtionally reset if it rebuilds from a code change, as well as from a data change.

Reply:
It has bad cache implications as runner has cache inside, but it's not a big deal.